service/service.go

```diff
@@ -31,38 +31,28 @@ import (
 // This functionality is EXPERIMENTAL and may be changed in later releases.
 
 type (
+	// Service represents a configured NATS service.
+	// It should be created using [Add] in order to configure the appropriate NATS subscriptions
+	// for request handler and monitoring.
+	Service struct {
+		// Config contains a configuration of the service
+		Config
 
-	// Service is an interface for service management.
-	// It exposes methods to stop/reset a service, as well as get information on a service.
-	Service interface {
```

Comment:
Why change away from an interface?

Reply:
I was wondering whether to better have interface or not, but my general reasoning is:
- we don't really need an interface here - unlike in JetStream simplification, where we'll probably want to support Push and Pull consumers separately (and possibly ordered consumer as well) on one interface, here we have 1 implementation
- by not using interfaces, we enable other users to add their own interfaces on top of `Service()` however they want, and we do not break anything for them if we want to add more methods
- core NATS client does not rely on any interface, so while it's not 100% consistent (`JesStreamContext` is an interface), we do not break conventions

Reply:
I think lately folks have pushed us more towards interfaces. I agree we should optimize for extensibility. 

@ripienaar What are your thoughts?

Reply:
If you want people to wrap it or augment with additional abilities without changing code then interfaces help.  So for KV for example its an interface and we can wrap a cache, codec, e2e encryption or whatever we like - those just wrap the interface and any other calling code doesnt need to change.

So if that kind of thing isnt a goal here then you can be more fluid about it. 

Regarding core nats package - Msg is a great example of someting that really should have been an interface. Rather than the way things are now with a Msg struct with a ton of methods on it thats just never used by core nats

Reply:
interface = can change internal details or have alternative implementations = win.